packages/components/pie-cookie-banner/test/component/pie-cookie-banner.spec.ts

```diff
@@ -149,47 +145,48 @@ test.describe('PieCookieBanner - Component tests', () => {
             },
         });
 
-        const cookieBanner = page.locator(componentSelector);
-
         // Act
-        await page.click(managePreferencesSelector);
-        await page.click(modalSaveButtonSelector);
-        const modal = page.locator(modalSelector);
+        await pieCookieBannerComponent.clickManagePreferencesAction();
+        await pieCookieBannerComponent.clickButtonWithText('Save');
         const [expectedCookieBannerPrefsSavedEvent] = preferences.filter(({ id }) => id !== 'all')
         .map(({ id, checked }) => ({ [id]: !!checked }));
 
+        const cookieBanner = await pieCookieBannerComponent.isCookieBannerVisible();
+        const modal = await pieModalComponent.isModalVisible();
+
         // Assert
-        expect(modal).not.toBeVisible();
-        expect(cookieBanner).not.toBeVisible();
+        expect(modal).toBeFalsy();
+        expect(cookieBanner).toBeFalsy();
         expect(cookieBannerPrefsSavedEvent).toMatchObject(expectedCookieBannerPrefsSavedEvent);
     });
 
-    test('should always toggle the `necessary` preference and set it to disabled', async ({ mount, page }) => {
+    test('should always toggle the `necessary` preference and set it to disabled', async ({ mount }) => {
         // Arrange
         await mount(PieCookieBanner, {
             props: {} as CookieBannerProps,
         });
 
         // Act
-        await page.click(managePreferencesSelector);
-        const necessaryPreferenceToggle = await page.locator(getPreferenceItemSelector('necessary'));
+        await pieCookieBannerComponent.clickManagePreferencesAction();
+        const necessaryPreferenceToggleChecked = await pieCookieBannerComponent.isPreferenceToggleChecked(getPreferenceItemSelector('necessary'));
```

Comment:
Is there any way this `getPreferenceItemSelector` could be worked into the cookie banner page object?

That would lead to something like this, which looks reasonable to me:
`pieCookieBannerComponent.isPreferenceToggleChecked('necessary')`